main.nf

```diff
@@ -11,64 +11,120 @@ nextflow.enable.dsl = 2
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    GENOME PARAMETER VALUES
+    IMPORT FUNCTIONS / MODULES / SUBWORKFLOWS / WORKFLOWS
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-params.fasta = WorkflowMain.getGenomeAttribute(params, 'fasta')
-params.dict = WorkflowMain.getGenomeAttribute(params, 'dict')
+include { VCFTOMAF  } from './workflows/vcftomaf'
+include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nfcore_vcftomaf_pipeline'
+include { PIPELINE_COMPLETION     } from './subworkflows/local/utils_nfcore_vcftomaf_pipeline'
+
+include { getGenomeAttribute      } from './subworkflows/local/utils_nfcore_vcftomaf_pipeline'
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    VALIDATE & PRINT PARAMETER SUMMARY
+    GENOME PARAMETER VALUES
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-include { validateParameters; paramsHelp } from 'plugin/nf-validation'
+params.fasta = getGenomeAttribute('fasta')
+params.dict = getGenomeAttribute('dict')
 
-// Print help message if needed
-if (params.help) {
-    def logo = NfcoreTemplate.logo(workflow, params.monochrome_logs)
-    def citation = '\n' + WorkflowMain.citation(workflow) + '\n'
-    def String command = "nextflow run ${workflow.manifest.name} --input samplesheet.csv --genome GRCh37 -profile docker"
-    log.info logo + paramsHelp(command) + citation + NfcoreTemplate.dashedLine(params.monochrome_logs)
-    System.exit(0)
-}
+// INTERVALS
+intervals = params.intervals    ? Channel.fromPath(params.intervals).collect()      : Channel.value([])
 
-// Validate input parameters
-if (params.validate_params) {
-    validateParameters()
-}
+// FASTA
+fasta        = params.fasta     ? Channel.fromPath(params.fasta).collect()          : Channel.value([])
+dict         = params.dict      ? Channel.fromPath(params.dict).collect()           : Channel.empty()
+
+// Chain file
+chain        = Channel.fromPath(params.chain).collect() //--> this fails  // params.chain     ? Channel.fromPath(params.chain).collect()          : Channel.empty()
```

Comment:
@mirpedrol any idea why this is not working here, but when I specify it in the call directly (see below) it works. Error is: `Missing process or function map([Script_59ab0145d1ad3f6e$_runScript_closure1$_closure2$_closure10@19240b9])` which I think comes from running `chain.map{}` in the workflow.nf.

Reply:
ok running `getClass` on chain and intervals gives:

```
intervals: class groovyx.gpars.dataflow.DataflowVariable
chain:      class nextflow.extension.OpCall
```

Reply:
resolved: chain should not be used, seems to be reserved